inference-engine/src/gna_plugin/optimizer/gna_pass_manager.cpp

```diff
@@ -268,6 +269,40 @@ void HandleMultipleActivationsForTheLayerPass::run() {
     }
 }
 
+void ForbidActivationFusingPass::run() {
+    for (auto& l : *pLayers) {
+        if (l->insData.empty()) continue;
+        auto prevLayer = CNNNetPrevLayerSkipCertain(l, 0, [](CNNLayerPtr ptr) {
+            return LayerInfo(ptr).isNonFunctional();
+            });
+        if (LayerInfo(l).isActivation()) {
+            if (LayerInfo(prevLayer).has32BOutput()) {
+                // find all layers directly connected to the outputs of the previous layer
+                const auto allUsingPrev = CNNNetGetAllNextLayersSkipCertain(prevLayer, -1,
+                    [&](CNNLayerPtr nextLayer) -> bool {
+                        for (const auto& input : nextLayer->insData) {
+                            for (const auto& output : prevLayer->outData) {
+                                if (areEqualDatas(input.lock(), output) &&
+                                    areEqualDatas(l->insData[0].lock(), output) &&
+                                    ((nextLayer->type == "Eltwise" && nextLayer->params.at("operation") == "sum")
```

Comment:
it is better to use LayerInfo and isEltwiseSum()

Reply:
Thanks, I will try with that

Reply:
It's done.